### ISSUE TITLE
[INFO] How Branching works for Packages

### ISSUE THREAD

**Issue by armab** [StackStorm/st2-packages#230]
`st2-packages` repo now follows `st2` branching strategy. What it means.

#### The Logic:
* Inside `st2-packages` repo:
  * pushing to `master` of `st2-packages` repo will generate packages based on `master` of `st2`
  * pushing to `v1.3` of `st2-packages` repo will generate packages based on `v1.3` of `st2`
  * opening PR against `master` branch of `st2-packages` repo will use `st2` `master` code
  * opening PR against `v1.3` branch of `st2-packages` repo will use `st2` `v1.3` code
  * opening PR against any other branch of `st2-packages` repo will use `st2` `master` code (fallback)
* Inside `st2` repo:
  * pushing to `master` of `st2` repo will generate packages based on `master` of `st2-packages`
  * pushing to `v1.3` of `st2` repo will generate packages based on `v1.3` of `st2-packages`

#### Rebuild the Package Manually: 
Normally pushing to `v1.3` branch for any `st2` or `st2-packages` repos will generate and deploy `v1.3` staging packages.
Usually that's enough, but if you want to do that manually without pushing changes, just use CircleCI rebuild:

- 1.) Navigate to `v1.3` branch builds https://circleci.com/gh/StackStorm/st2-packages/tree/v1.3
- 2.) Find latest build
- 3.) Click `Rebuild`

**Comment by armab** [comment]
^^ cc @enykeev @dennybaa @lakshmi-kannan for logic review.


That'll be better automated after merging: 
https://github.com/StackStorm/st2-packages/pull/227
https://github.com/StackStorm/st2-packages/pull/229
https://github.com/StackStorm/st2/pull/2569
https://github.com/StackStorm/st2/pull/2570

**Comment by enykeev** [comment]
> opening PR against any other branch of st2-packages repo will use st2 master code (fallback)

Not really. As of https://github.com/StackStorm/st2-packages/commit/86a4a43dec0db8d8815ab1cfdd226dab26bdb6dc, it depends on the point you're branching from. Any commit that branches after 86a4a43dec0db8d8815ab1cfdd226dab26bdb6dc will use 1.3 branch of st2

**Comment by armab** [comment]
@enykeev see: https://github.com/StackStorm/st2-packages/pull/227

**Comment by armab** [comment]
Yep, as discussed with @enykeev, closing #227 as error-prone way.
